src/main/java/racingcar/domain/Car.java

```diff
@@ -15,4 +24,36 @@ public Distance move(Moving moving) {
         }
         return total;
     }
+
+    public String name() {
+        return name;
+    }
+    
+    public int distance() {
+        return total.getDistance();
+    }
+
+    @Override
+    public int compareTo(Car car) {
+        if (total.greaterThan(car.total)) {
+            return GREATER;
+        }
+        if (total.equals(car.total)) {
+            return EQUALS;
+        }
+        return LESS;
+    }
```

Comment:
결국 Distance의 값을 비교하는거니 Distance 내부로 이 로직을 옮겨볼 수도 있을 것 같아요 😃

Reply:
@testrace 요거는 아래의 `Cars` 에서 가장 주행거리가 긴 차 순으로 내림차순 정렬하는 코드입니당
```
public Cars longest() {
        Car max = Collections.max(cars);
        List<Car> longest = new ArrayList<>();
        for (Car car : cars) {
            addEqualsDistance(max, longest, car);
        }
        return new Cars(longest);
    }
```

`Car max = Collections.max(cars);` max() 에서 가장 긴 거리를 주행한 차를 리턴받기 위해서 해당 메소드를 Car 에 구현하였어용!
요렇게 안하면 Cars 에서 직접 distance 를 비교해가며 가장 주행거리가 긴 차를 찾는 코드를 직접 구현해야 해서 요걸 사용해봤는데 혹시 요부분에서 조언해주실 부분이 있을까용??

Reply:
Distance 클래스도 Comparable 인터페이스를 구현하고 있다면
Car에서는 로직이 조금 더 간견해 질 수 있을 것 같아서 의견 드렸습니다 😃 

값 자체를 비교하는 로직을 Distance가 가진다면 아래와 같이 구현할 수 있을 것 같아요!

```suggestion
    @Override
    public int compareTo(Car car) {
        return this.total.compareTo(car.total);
    }
```